fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/AgilityContext.java

```diff
@@ -358,20 +385,22 @@ public void flushAndClose() {
         }
 
         @Override
-        public void abortAndReset() {
+        public void abortAndClose() {
             // Here: the lock status is undefined. The main goal of this function is to revive this object for post failure cleanups
             synchronized (commitLockSync) {
+                closed = true;
+                committingNow = true; // avoid future commits
+                currentWriteSize = 0;
+                if (currentContext != null) {
+                    currentContext.close();
+                    currentContext = null;
+                }
+                // release the locks, in case another thread is waiting on them
```

Comment:
This will just mean that the other operations will fail fast, right?

Reply:
New read/writes will fail in `ensureOpen()`. If there is any commitNow() waiting for a write lock, it will become a no-op. 
Added an `ensureOpen()` to `createIfNeeded()` in case there is a user operation waiting for a read lock. 
✅